src/backend/storage/smgr/smgr.c

```diff
@@ -175,6 +174,18 @@ smgrshutdown(int code, Datum arg)
 	}
 }
 
+#define SMGR_CHAIN_LOOKUP(SMGR_METHOD) while(chain_index < reln->smgr_chain.size && smgrsw[reln->smgr_chain.chain[chain_index]].SMGR_METHOD == NULL) chain_index++
+#define SMGR_CHAIN_CHECK_END_ASSERT() Assert(chain_index < reln->smgr_chain.size)
```

Comment:
Should we have a check somewhere that the method we ended up isn't NULL? In case of all the chain has NULL for create, for example. As far as I can see there is a special case for shutdown but the rest of the methods should be implemented at least at the end of the chain

Reply:
Yes, I'll add a check for "tail" smgrs.